### DISCUSSION TITLE
Need clarification of some of the source files

### DISCUSSION THREAD

**Discussion by Yevgnen** [melih-unsal/DemoGPT#12] [Q&A] [answered]
I'm confused by the files like `src/langchain_agent.py`, `src/langchain_code.py`, `src_langchain_expert.py`, as they contain unmentioned dependencies like `tqdm` and `fire`.

Also how are the `app.py`, `model.py` and `prompt.py` in `src/` different from those in `src/prompt_based`?

I don't seem to find related information on the documents. Are these file deprecated? If so, it'd better to remove them and keep them only in the version history.

Thanks.

**Comment by melih-unsal** [comment, marked as answer]
Hi @Yevgnen , I really appreciate your attention.
Actually I'm close to new version 1.2 which will be capable of generating broader chains from prompt and these files are corresponding to that upcoming release. Within a week, they will be the new version's main files. I shouldn't put them in the main branch but this was the real reason 🙏🏻
